controllers/storagecluster/cephobjectstores.go

```diff
@@ -175,6 +186,28 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 			r.Log.Error(err, "Failed to set ControllerReference for CephObjectStore.", "CephObjectStore", klog.KRef(obj.Namespace, obj.Name))
 			return nil, err
 		}
+		// if kmsConfig is not 'nil', add the KMS details to CephCluster spec
+		if kmsConfigMap != nil {
+			// Set default KMS_PROVIDER and VAULT_SECRET_ENGINE values, refer https://issues.redhat.com/browse/RHSTOR-1963
+			if _, ok := kmsConfigMap.Data["KMS_PROVIDER"]; !ok {
```

Comment:
Is there a scenario where `KMS_PROVIDER` does not exist?

Reply:
I have copied from [CephClusterSpec] (https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/cephcluster.go#L402)

Reply:
Ok.